egs/librispeech/ASR/transducer_stateless/decoder.py

```diff
@@ -86,13 +88,20 @@ def forward(self, y: torch.Tensor, need_pad: bool = True) -> torch.Tensor:
         if self.context_size > 1:
             embeding_out = embeding_out.permute(0, 2, 1)
             if need_pad is True:
-                embeding_out = F.pad(
-                    embeding_out, pad=(self.context_size - 1, 0)
-                )
+                if self.backward:
+                    assert self.context_size == 2
+                    embeding_out = F.pad(
+                        embeding_out, pad=(0, self.context_size - 1)
+                    )
+                else:
+                    embeding_out = F.pad(
+                        embeding_out, pad=(self.context_size - 1, 0)
+                    )
```

Comment:
I don't understand why it can use the next two symbols to predict the current one just by changing the padding direction.

Reply:
If the input is [sos, a, b, c, d] and output is [a, b, c, d, eos], padding left and using kernel-size 2, it uses left context. If the input is [a, b, c, d, eos] and output is [sos, a, b, c, d], padding right and using kernel-size 2, I think it uses right context.

Reply:
Maybe @danpovey can have a look at this implementation. As the kernel size is 2, the i-th output symbol always sees only the (i-1)-th and i-th symbols. It never uses its next neighbors if the input sequence is not reversed.

Reply:
@csukuangfj I think it takes care of at least part of the shift needed when he does:
```eos_y_padded = F.pad(eos_y_padded[:, 1:], pad=(0, 1), value=blank_id)```
... i.e. the 1st element of the output (y) is discarded.  I'm not crazy about the overall structure or documentation
of this code (I know we pulled it from somewhere).  Kangwei, also I'm not quite sure what you mean by "input" or "output" here.

Reply:
The "input" means the sequences feeding into the decoder, "output" means the labels used to calculate loss value.  Sorry for the confusion, will update the code and documention.